UnitsNet.Tests/CustomCode/LengthTests.FeetInches.cs

```diff
@@ -29,63 +31,121 @@ public void FeetInchesRoundTrip()
+            new object[]{"-1' 1\"", -1.08333333, new CultureInfo("de-CH")},     // Normal form
+            new object[]{"-1′ 1″", -1.08333333, new CultureInfo("de-CH")},      // Normal form
+            new object[]{" -1′ 1″ ", -1.08333333, new CultureInfo("de-CH")},    // Normal form, requires trimming
+            new object[]{"-1'1\"", -1.08333333, new CultureInfo("de-CH")},      // Without space
+            new object[]{"-1′1″", -1.08333333, new CultureInfo("de-CH")},       // Without space
+            new object[]{"-1 ft 1 in", -1.08333333, new CultureInfo("de-CH")},
+            new object[]{"-1ft 1in", -1.08333333, new CultureInfo("de-CH")},
+        };
+
         [Theory]
-        [InlineData("1'", 1)] // Feet only
-        [InlineData("1′", 1)] // Feet only
-        [InlineData("1\"", 0.08333333)] // Inches only
-        [InlineData("1″", 0.08333333)] // Inches only
-        [InlineData("0' 1\"", 0.08333333)] // Inches only
-        [InlineData("0' 1″", 0.08333333)] // Inches only
-        [InlineData("0′ 1\"", 0.08333333)] // Inches only
-        [InlineData("0′ 1″", 0.08333333)] // Inches only
-        [InlineData("1' 1\"", 1.08333333)] // Normal form
-        [InlineData("1′ 1″", 1.08333333)] // Normal form
-        [InlineData(" 1′ 1″ ", 1.08333333)] // Normal form, requires trimming
-        [InlineData("1'1\"", 1.08333333)] // Without space
-        [InlineData("1′1″", 1.08333333)] // Without space
-        [InlineData("1 ft 1 in", 1.08333333)]
-        [InlineData("1ft 1in", 1.08333333)]
-        [InlineData("-1'", -1)] // Feet only
-        [InlineData("-1′", -1)] // Feet only
-        [InlineData("-1\"", -0.08333333)] // Inches only
-        [InlineData("-1″", -0.08333333)] // Inches only
-        [InlineData("-0' 1\"", -0.08333333)] // Inches only
-        [InlineData("-0' 1″", -0.08333333)] // Inches only
-        [InlineData("-0′ 1\"", -0.08333333)] // Inches only
-        [InlineData("-0′ 1″", -0.08333333)] // Inches only
-        [InlineData("-1' 1\"", -1.08333333)] // Normal form
-        [InlineData("-1′ 1″", -1.08333333)] // Normal form
-        [InlineData(" -1′ 1″ ", -1.08333333)] // Normal form, requires trimming
-        [InlineData("-1'1\"", -1.08333333)] // Without space
-        [InlineData("-1′1″", -1.08333333)] // Without space
-        [InlineData("-1 ft 1 in", -1.08333333)]
-        [InlineData("-1ft 1in", -1.08333333)]
-        public void TryParseFeetInches(string str, double expectedFeet)
+        [MemberData(nameof(ValidData))]
+        public void TryParseFeetInches(string str, double expectedFeet, CultureInfo formatProvider)
         {
-            Assert.True(Length.TryParseFeetInches(str, out Length result));
+            Assert.True(Length.TryParseFeetInches(str, out Length result, formatProvider));
             AssertEx.EqualTolerance(expectedFeet, result.Feet, 1e-5);
         }
 
+        public static IEnumerable<object[]> InvalidData => new List<object[]>
```

Comment:
If you want, we could refactor this to combinatorial parameters. It would make it a lot less verbose to cover all the same test cases.

https://github.com/AArnott/Xunit.Combinatorial

Reply:
Oh man. I Googled for it using the word "matrix" as was bummed thinking you couldn't do it. Thanks!

Reply:
Matrix was my first go to search as well :D

Reply:
... and I implemented it first in a do-it-yourself way (see patch below)